Composer/packages/extension/src/storage/store.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import path from 'path';
+
+import { existsSync, writeJsonSync, readJsonSync, unlinkSync } from 'fs-extra';
+
+export type IStore<T> = {
+  read: () => T;
+  write: (data: T) => void;
```

Comment:
@benbrown would you prefer this interface take a key when reading and writing?

```ts
type IStore {
  read: (key: string) => unkown;
  write<T>: (key: string, value: T) => void
}
```

Reply:
yes
read write delete? or write null to delete?